src/logic/fetch-elements/fetch-third-party-wearables.ts

```diff
@@ -101,52 +158,176 @@ export async function fetchUserThirdPartyAssets(
 
   const thirdPartyId = parts.slice(0, 5).join(':')
 
-  let thirdPartyProvider: ThirdPartyProvider | undefined = undefined
-
   const thirdPartyProviders = await components.thirdPartyProvidersStorage.getAll()
-  for (const provider of thirdPartyProviders) {
-    if (provider.id === thirdPartyId) {
-      thirdPartyProvider = provider
-      break
-    }
-  }
+  const thirdPartyProvider: ThirdPartyProvider | undefined = thirdPartyProviders.find(
+    (provider) => provider.id === thirdPartyId
+  )
 
   if (!thirdPartyProvider) {
     return []
   }
 
-  const assetsByOwner = await fetchAssets(components, owner, thirdPartyProvider)
-  if (!assetsByOwner) {
-    throw new Error(`Could not fetch assets for owner: ${owner}`)
-  }
+  const thirdPartyWearables = await doTheWork(components, owner, [thirdPartyProvider])
 
-  return assetsByOwner.filter((asset) => asset.urn.decentraland.startsWith(thirdPartyId)) ?? []
+  return thirdPartyWearables.map((tpw) => ({
+    id: tpw.urn, // TODO check this, not sure id refers to full urn, it might be provider + collection id + item id
+    amount: tpw.amount,
+    urn: {
+      decentraland: tpw.urn
+    }
+  }))
 }
 
 export async function fetchAllThirdPartyWearables(
-  components: Pick<AppComponents, 'thirdPartyProvidersStorage' | 'fetch' | 'logs' | 'entitiesFetcher' | 'metrics'>,
+  components: Pick<
+    AppComponents,
+    | 'alchemyNftFetcher'
+    | 'contentServerUrl'
+    | 'thirdPartyProvidersStorage'
+    | 'fetch'
+    | 'logs'
+    | 'entitiesFetcher'
+    | 'metrics'
+  >,
   owner: string
 ): Promise<ThirdPartyWearable[]> {
   const thirdParties = await components.thirdPartyProvidersStorage.getAll()
 
-  // TODO: test if stateValue is kept in case of an exception
-  const thirdPartyAssets = (
-    await Promise.all(thirdParties.map((thirdParty: ThirdPartyProvider) => fetchAssets(components, owner, thirdParty)))
-  ).flat()
-
-  const entities = await components.entitiesFetcher.fetchEntities(thirdPartyAssets.map((tpa) => tpa.urn.decentraland))
-  const results: (ThirdPartyAsset & { entity: Entity })[] = []
-  for (let i = 0; i < thirdPartyAssets.length; ++i) {
-    const entity = entities[i]
-    if (entity) {
-      results.push({
-        ...thirdPartyAssets[i],
-        entity
-      })
+  return await doTheWork(components, owner, thirdParties)
+}
+
+async function doTheWork(
```

Comment:
Lovely name ❤️ 
WDYT about renaming it to `_fetchThirdPartyWearables` or `_fetchAllThirdPartyWearables`?

Reply:
😂  Done.